src/lib/mina.ts

```diff
@@ -57,6 +54,68 @@ interface Mina {
   getAccount(publicKey: PublicKey): Account;
 }
 
+function createUnsignedTransaction(f: () => unknown) {
+  return createTransaction(undefined, f);
+}
+
+function createTransaction(sender: PrivateKey | undefined, f: () => unknown) {
+  if (currentTransaction !== undefined) {
+    throw new Error('Cannot start new transaction within another transaction');
+  }
+
+  currentTransaction = {
+    sender,
+    parties: [],
+    nextPartyIndex: 0,
+  };
+
+  try {
+    Circuit.runAndCheckSync(f);
+  } catch (err) {
+    currentTransaction = undefined;
+    // TODO would be nice if the error would be a bit more descriptive about what failed
+    throw err;
+  }
+
+  const otherParties: Array<Party_> = currentTransaction.parties.map((party) =>
+    toParty(party)
+  );
+
+  let feePayer: FeePayerParty;
+  if (sender !== undefined) {
+    // if sender is provided, fetch account to get nonce
+    const senderPubkey = sender.toPublicKey();
+    let senderAccount = getAccount(senderPubkey);
+    feePayer = {
+      body: toPartyBody(Body.keepAll(senderPubkey)),
+      predicate: senderAccount.nonce,
+    };
+  } else {
+    // otherwise use a dummy fee payer that has to be filled in later
+    feePayer = {
+      body: toPartyBody(Body.dummy()),
+      predicate: UInt32.zero,
+    };
+  }
+
+  const txn: Parties = { otherParties, feePayer };
+
+  nextTransactionId.value += 1;
+  currentTransaction = undefined;
```

Comment:
we should do something similar w.r.t. getting rid of globals with this value -- same thing though as my note on the prior PR; we can make an issue if it's not an obvious fix

Reply:
I made one: https://github.com/o1-labs/snarkyjs/issues/134